cloudify_cli/env.py

```diff
@@ -459,20 +460,35 @@ def get_auth_header(username, password):
 
 
 class ClusterHTTPClient(HTTPClient):
+    timeout = 5
+
     def __init__(self, *args, **kwargs):
         super(ClusterHTTPClient, self).__init__(*args, **kwargs)
         if not profile.cluster:
             raise ValueError('Cluster client invoked for an empty cluster!')
         self._cluster = list(profile.cluster)
 
     def do_request(self, *args, **kwargs):
-        for node in profile.cluster:
+        # this request can be retried for each manager - if the data is
+        # a generator, we need to copy it, so we can send it more than once
+        copied_data = None
+        if isinstance(kwargs.get('data'), types.GeneratorType):
+            copied_data = itertools.tee(kwargs.pop('data'),
+                                        len(profile.cluster))
+
+        if kwargs.get('timeout') is None:
+            kwargs['timeout'] = self.timeout
```

Comment:
instead of last 2 lines, can use: kwargs.setdefault('timeout', self.timeout)
or kwargs.setdefault('timeout', self.default_timeout_sec)